code/modules/events/koi_mirgration.dm

```diff
@@ -0,0 +1,41 @@
+/datum/event/koi_migration
+	announceWhen	= 50
+	endWhen 		= 900
+
+	var/list/spawned_koi = list()
+
+/datum/event/koi_migration/setup()
+	announceWhen = rand(40, 60)
+	endWhen = rand(600,1200)
+
+/datum/event/koi_migration/announce()
```

Comment:
Is this just copy paste from space carp @Aurorablade isss itttttt.

Reply:
Pretty much. Sorry for the lazyness, considering what i wanted to do existed in some fashion i wasn't sure what would make sense to keep remove.

Reply:
if you're just going to do that, then please make it more OOP.

Reply:
To make this more OOP, you could refactor the carp migration to have a couple extra vars, or perhaps a weighted list, containing mob paths to spawn during the event.
And then just implement the koi migration as a list of one item, the koi mob path.